### DISCUSSION TITLE
filter or XPath, which is faster?

### DISCUSSION THREAD

**Discussion by pr-apes** [speedata/publisher#495] [Q&A] [answered]
@pgundlach,

I have a layout that I use to rearrange pages from PDF files in different forms based on their file name and page size.

Since it was easier for me (and `matches()` was added recently), I wrote a Lua filter to invoke modes parsing the file name patterns.

I wonder whether passing a variable with the filename to the layout and parsing them with `matches()` could be faster than loading a filter.

Which approach do you think it could be faster: Lua filter or XPath `matches()`?

Many thanks for your help.

**Comment by pgundlach** [comment, marked as answer]
I doubt that there is much difference. matches might be a bit slower due to a Go-C-bridge (the Lua backend calls a C function which uses a library call  written in Go) but the filter might be slower because the Lua code might not be optimal. So I don't have a feeling which I'd recommend.

Sorry for the no-answer.

**Reply by pr-apes** [reply]
Ok, I will try the XPath approach and measure the difference.
